app/src/main/java/com/github/polypoly/app/network/StorableObject.kt

```diff
@@ -95,7 +131,7 @@ abstract class StorableObject<T : Any> (dbPath: String, val key: String) {
      *
      * @note /!\ STATIC /!\ The returned value shouldn't depend on the current object instance
      */
-    protected abstract fun toLocalObject(dbObject: T): StorableObject<T>
+    protected abstract fun toLocalObject(dbObject: T): CompletableFuture<StorableObject<T>>
```

Comment:
good one-line example of why having the infrastructure + the code that uses it in a single PR can be useful :-)